src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisSettingsResource.java

```diff
@@ -0,0 +1,174 @@
+package de.tum.in.www1.artemis.web.rest.iris;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.iris.settings.IrisSettings;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.security.annotations.EnforceAdmin;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.iris.IrisSettingsService;
+
+/**
+ * REST controller for managing {@link IrisSettings}.
+ */
+@RestController
+@RequestMapping("api/")
+public class IrisSettingsResource {
+
+    private final Logger log = LoggerFactory.getLogger(IrisSettingsResource.class);
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final IrisSettingsService irisSettingsService;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final ProgrammingExerciseRepository programmingExerciseRepository;
+
+    public IrisSettingsResource(UserRepository userRepository, CourseRepository courseRepository, IrisSettingsService irisSettingsService,
+            AuthorizationCheckService authCheckService, ProgrammingExerciseRepository programmingExerciseRepository) {
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+        this.irisSettingsService = irisSettingsService;
+        this.authCheckService = authCheckService;
+        this.programmingExerciseRepository = programmingExerciseRepository;
+    }
+
+    /**
+     * GET iris/global-iris-settings: Retrieve the raw iris settings for the course.
+     *
+     * @return the {@link ResponseEntity} with status {@code 200 (Ok)} and with body the settings.
+     */
+    @GetMapping("iris/global-iris-settings")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
```

Comment:
doesn't follow our authorization guidelines
should be @EnforceAtLeastInstructor